wagtail/wagtailcore/tests/test_blocks.py

```diff
@@ -175,6 +175,39 @@ def test_validate_non_required_richtext_block(self):
         self.assertIsInstance(result, RichText)
         self.assertEqual(result.source, '')
 
+    def test_widget_field_richtext_block(self):
+        from wagtail.wagtailcore.fields import RichTextArea
+        custom_widget = RichTextArea(editor_config={'plugins': {'halloheadings': {'formatBlocks': ['p', 'h2']}}})
+        block = blocks.RichTextBlock(widget=custom_widget)
+        self.assertEqual(block.field_options.get('widget'), custom_widget)
+
+    def test_widget_custom_widget_richtext_block(self):
+        from wagtail.wagtailcore.fields import BaseTextAreaWidget
+
+        class CustomTextArea(BaseTextAreaWidget):
+            def get_panel(self):
+                pass
+        custom_widget = CustomTextArea(editor_config={'custom': {'testix': {'formatBlocks': ['p', 'h2']}}})
+        block = blocks.RichTextBlock(widget=custom_widget)
+
+        self.assertIsInstance(block.field_options.get('widget'), CustomTextArea)
+        self.assertIsInstance(block.field.widget, CustomTextArea)
+
+    def test_widget_default_widget_richtext_block(self):
+        from wagtail.wagtailcore.fields import RichTextArea
+        deafult_widget = RichTextArea()
+        block = blocks.RichTextBlock(widget=deafult_widget)
```

Comment:
If we want to test default widget, we must not to pass `widget` argument to `RichTextBlock`.

I'm not sure about checks. Maybe something like this this?
```python
    def test_widget_default_widget_richtext_block(self):
        block = blocks.RichTextBlock()

        self.assertIsInstance(block.field.widget, RichTextArea)
```